content/05.model-complexity.md

```diff
@@ -1,31 +1,32 @@
 ## Reducing misinterpretation of model output with statistical techniques
 
-ML generally works well on data that meet a few critical assumptions.
+The successful application of ML can be improved by meeting certain conditions.
 First, the dataset contains an equal number of samples for all categories (no “class imbalance”).
```

Comment:
Justin: Don’t agree with the characterization. Balance is not required; more important is sufficient representation of each category, capturing the biological variability for that class

Reply:
```suggestion
First, assuming that all categories in a dataset have comparable variance, the dataset contains an equal number of samples for all categories (no “class imbalance”).
```

Reply:
i think the takeaway that we should include here is different than what we have currently written. Class imbalance is a problem in rare disease less so because of the training, and more so in the testing/evaluation. That is, if you have a classifier to identify ultra rare disease vs non ultra rare disease people, it can be very accurate (e.g. >0.99) if it just guesses that everyone does not have the disease. 

I think the takeaway here should be (1) you need to have a large enough number of rare disease observations to capture the variation in features associated with that rare disease and (2) that evaluation metrics need to be carefully considered depending on the question being asked. The takeaways aren't really rare disease specific, but the fact that class imbalance will exist simply by virtue of looking at rare disease is somewhat rare disease specific.  

what do you all think?

Reply:
Agree with both (1) and (2). I think we make the case for (1) in the heterogeneity section. We have not explicitly stated (2) but the best place for that may be the outlook section.

This section mostly delves into the statistical assumptions that an ML method has, given that originally they are developed for large datasets. This section later goes into examples of statistical techniques like ensemble methods and regularization which help generate synthetic data points and reduce feature space to suit the sample space better respectively.

Reply:
Agree with both (1) and (2) as well. However, I don't think (1) is sufficiently captured in the heterogeneity section as currently written. (2) should probably at least get mentioned in Box 2 in addition to the outlook section.

I will have a PR to this branch that addresses the latter. This sentence should get re-written to address (1) in my opinion.

Reply:
Maybe?

```suggestion
First, the dataset contains sufficient representation from each class such that relevant variability from that class is captured.
```

And then another sentence that ties this to class imbalance.

I am not successfully reducing words here, but do we absolutely need to talk about class imbalance in this paragraph? Would it be sufficient to talk about it later in the context of what is now line 24?

Reply:
I am a bit late here but I think the current version looks pretty good! I will be interested to see what Justin thinks.